tensorflow_datasets/scripts/random_image_generator.py

```diff
@@ -0,0 +1,107 @@
+
+""" Generate random images that compress better.
+By replacing the larger images with more compressable equivalents."""
+
+import os
+from zipfile import ZipFile
+from PIL import Image
+import numpy as np
+import tarfile
+import zipfile
+import shutil
+import gzip
+
+
+# images_path variable to store the path of images for later compression
+images_path = []
+
+
+def image_process(root):
+    """ Generate random images and remove noise of the
+        images to compress better.
+
+          Args:
+           root: path of the images to get processed
+
+    """
+    images_path.append(root)
+    image = np.array(Image.open(root))
+    image = np.ones_like(image%255) * np.random.randint(np.amax(image%255))
```

Comment:
I'm not sure what you're trying to compute here. To get deterministic value, you shouldn't use `np.random.randint` anymore as it will compute a new value each time the script is executed.

We want a deterministic mapping `filepath -> integer` to generate the grey_scale between 0 and 255.
and then compress the image like:

1) Extract the relative tfds path from the `filepath` to relative_filepath.
Ex: `/home/user/xyz/tensorflow_dataset/some_path/file.jpg` -> `some_path/file.jpg`

2) Generate the integer by hashing the file path
```
grey_scale = hash_fn(relative_filepath) % 255
```
3) Create the new image.
```
image = np.ones_like(image) * `grey_scale`
```